src/pineko/kfactor.py

```diff
@@ -384,35 +304,41 @@ def compute_k_factor_grid(
         kfactors folder
     yamldb_path : pathlib.Path()
         path to the yaml file describing the dataset
-    max_as : int
-        max as order
+    pto_to_update : int
+        perturbative order to update: 1 = LO, 2 = NLO ...
+        no matter which power of alpha_s it is.
     target_folder: pathlib.Path
-        path where store the new grid (optional)
+        path where store the new grid
+    order_exists: bool
+        True if the order to update is already present
     """
     import lhapdf  # pylint: disable=import-error,import-outside-toplevel
 
-    # With respect to the usual convention here max_as is max_as-1
-    max_as_test = max_as - 1
     # Extracting info from yaml file
     with open(yamldb_path, encoding="utf-8") as f:
         yamldict = yaml.safe_load(f)
+
+    # loop on operands
     for grid_list in yamldict["operands"]:
+        # loop on grids
         for grid in grid_list:
+            # TODO: generalize for other type of kfactors ?
             cfac_path = kfactor_folder / f"CF_QCD_{grid}.dat"
             if "ATLASDY2D8TEV" in grid:
                 cfac_path = kfactor_folder / f"CF_QCDEWK_{grid}.dat"
-            centrals_kfactor, pdf_set = read_kfactor(cfac_path)
+
+            grid_name = f"{grid}.pineappl.lz4"
+            current_grid = pineappl.grid.Grid.read(grids_folder / grid_name)
+
+            central_kfactor, pdf_set = read_kfactor(cfac_path)
+            central_kfactor_filtered = to_list(current_grid, central_kfactor)
             alphas = lhapdf.mkAlphaS(pdf_set)
-            grid_path = grids_folder / (f"{grid}.pineappl.lz4")
-            pigrid = pineappl.grid.Grid.read(grid_path)
-            centrals_kfactor_filtered = filter_k_factors(pigrid, centrals_kfactor)
+
             do_it(
```

Comment:
```suggestion
            apply_to_grid(
```

Reply:
done in 4665cadc6dfd9f262b5b490504159becbe3656fa